lib/Runtime/Library/JavascriptArray.inl

```diff
@@ -955,45 +955,45 @@ SECOND_PASS:
             return true;
         }
 
-        SparseArraySegment<T> *toSegment = PrepareSegmentForMemOp<T>(toStartIndex, length);
-        if (toSegment == nullptr)
+        const auto isSegmentValid = [length](Js::SparseArraySegment<T>* segment, uint32 startIndex) {
+            uint32 end, segmentEnd;
+            // Check the segment is long enough
+            return (
+                segment &&
+                !UInt32Math::Add(startIndex, length, &end) &&
+                !UInt32Math::Add(segment->left, segment->length, &segmentEnd) &&
```

Comment:
No because `PrepareSegmentForMemOp` should modify the length of the segment to be long enough for the copy.
That particular check could actually be an assert for the dst, but I figured it was safer to do the check.

I still don't know what happens if this check fails since the segment has already been modified. 
I don't know the `PrepareSegmentForMemOp` code and since there has been no bug with it so far, I assume it correctly prepares the segment (or return `nullptr` which is fine to bailout then)
The only problematic case is if `PrepareSegmentForMemOp` returns a segment that is not big enough, which I think cannot happen